legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/PlanExecutor.java

```diff
@@ -169,6 +170,9 @@ public Result execute(SingleExecutionPlan singleExecutionPlan, ExecutionState st
             if (singleExecutionPlan.authDependent) {
                 state.setAuthUser((singleExecutionPlan.kerberos == null) ? user : singleExecutionPlan.kerberos);
             }
+            if (state.authId == null) {
+                state.setAuthUser(IdentityFactoryProvider.getInstance().makeIdentity(profiles).getName(), false);
+            }
             singleExecutionPlan.getExecutionStateParams(org.eclipse.collections.api.factory.Maps.mutable.empty()).forEach(state::addParameterValue);
```

Comment:
Can line 1761 be the same as 174 ? atleast wrto computing the current user ?

Reply:
Yes, this will evaluate to the same value wrto. evaluating current user. In the case _kerberos_ or _user_ is non null, this will evaluate to the Kerberos value we expect.

Reply:
If authDependent, the kerberos is bound to the plan while IdentityFactory evaluates the profile id at runtime. For current_user, we don't care about this as much since semantically we could support "current" as being extracted from the plan or the profile. We didn't have a plan previously since we didn't support this in Engine, so this is up to us to define if reasonable.